inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp

```diff
@@ -93,20 +93,58 @@ void FrontEnd::parsePad(const Model& model, const ie::CNNLayerPtr& _layer, const
     auto layer = std::dynamic_pointer_cast<ie::PadLayer>(_layer);
     IE_ASSERT(layer != nullptr);
 
-    IE_ASSERT(layer->pads_begin.size() == 4);
-    IE_ASSERT(layer->pads_end.size() == 4);
+    const auto ndims = inputs[0]->desc().dimsOrder().numDims();
+    VPU_THROW_UNLESS(ndims == 3 || ndims == 4, "Layer %s support only 3D and 4D input", layer->name);
+
+    IE_ASSERT(layer->pads_begin.size() <= 4);
+    IE_ASSERT(layer->pads_end.size() <= 4);
+
+    unsigned int pads_beginN = 0;
+    unsigned int pads_beginC = 0;
+    unsigned int pads_beginH = 0;
+    unsigned int pads_beginW = 0;
+
+    unsigned int pads_endN = 0;
+    unsigned int pads_endC = 0;
+    unsigned int pads_endH = 0;
+    unsigned int pads_endW = 0;
+
+    switch (ndims) {
+        case 4:
```

Comment:
It is ok, but I think we can come up with a more elegant solution :)

Reply:
I prepared more elegant solution, please review one more time

Reply:
Thanks, looks great!